src/admin/admin.js

```diff
@@ -0,0 +1,334 @@
+
+        if (selectedIndex === -1) {
+            newSelected = newSelected.concat(selected, id);
+        } else if (selectedIndex === 0) {
+            newSelected = newSelected.concat(selected.slice(1));
+        } else if (selectedIndex === selected.length - 1) {
+            newSelected = newSelected.concat(selected.slice(0, -1));
+        } else if (selectedIndex > 0) {
+            newSelected = newSelected.concat(
+                selected.slice(0, selectedIndex),
+                selected.slice(selectedIndex + 1),
+            );
+        }
+
+        setSelected(newSelected);
+    };
+
+    const isSelected = id => selected.indexOf(id) !== -1;
+
+    const fetchMonth = async (month, year) => {
+        setIsLoading(true);
+        rows = [];
+
+        const db = firebase.firestore();
+        const response = await db
+            .collection("months")
+            .where("month", "==", month)
+            .where("year", "==", year)
+            .get();
+        rows = response.docs.map(doc => {
+            const row = doc.data();
+            row.id = doc.id;
+            initRow(row);
+            return row;
+        });
+        setIsLoading(false);
+    };
+
+    const initRow = row => {
+        Constants.columns.forEach(cell => {
+            if (cell.numeric) {
+                row[cell.id] = getTotalHoursPerCategory(row.days, cell.id);
+            }
+        });
+    };
+
+    const getTotal = column => {
+        if (!column.numeric) {
+            return "";
+        }
+
+        const values = rows.map(x => x[column.id]);
+        return values.reduce((total, currentValue) => {
+            return Number(total) + Number(currentValue);
+        });
+    };
+
+    React.useEffect(() => {
+        fetchMonth(currentMonth, currentYear);
+    }, [currentMonth, currentYear]);
+
+    return (
+        <div className={classes.root}>
+            <Paper className={classes.paper}>
+                <OverviewToolbar
+                    numSelected={selected.length}
+                    currentMonth={currentMonth}
+                    currentYear={currentYear}
+                    onChangeDate={handleChangeDate}
+                />
+
+                <TableContainer>
+                    <Table
+                        className={classes.table}
+                        aria-labelledby="tableTitle"
+                        size="small"
+                        aria-label="enhanced table"
+                    >
+                        <OverviewHeader
+                            classes={classes}
+                            numSelected={selected.length}
+                            order={order}
+                            orderBy={orderBy}
+                            onSelectAllClick={handleSelectAllClick}
+                            onRequestSort={handleRequestSort}
+                            rowCount={rows.length}
+                        />
+                        {!isLoading && rows.length > 0 && (
```

Comment:
Hier zou je een eigen component van kunnen maken die zelf checkt of hij wel/niet nodig is. Dat maakt dit bestand ook wat kleiner en overzichtelijker.

Reply:
Doe ik in een volgende slag :)